### PR TITLE
Issue #454: Fix issue with directory not being found during cd (chang…

### PR DESCRIPTION
**Pull request by adchilds**
…e directory) if path contained spaces.

This fixes the issue noted in https://github.com/shockone/black-screen/issues/454 though I'm not sure how updating this function may affect the application as a whole, so input is needed.

### CODE REVIEW

File: src/utils/Common.ts [hunk 1]
```diff
@@ -181,7 +181,22 @@ export function resolveDirectory(pwd: string, directory: string): string {
...
+    return Path.resolve(pwd, parsedPath);
+}
+
+/**
+ * Replace any special characters to create a fully qualified path.
+ *
+ * '~' - the user's home directory
+ * '\' - escapes space character in file/directory path
+ */
+export function sanitizeSpecialCharacters(path: string): string {
```
Comment (adchilds) [inline src/utils/Common.ts]: This function technically doesn't need to be exported. Let me know if that's an issue.

**Comment by shockone** [comment]
Hey, @adchilds. This is a good contribution! But I'm afraid this place is too late to fix this issue. Stripping special characters is a job of a lexer. If you'd like to redo the PR, you'd need a few things:

* Change the `lex` function to throw away those backslashes https://github.com/shockone/black-screen/blob/master/src/CommandExpander.ts#L28-L28
* Change a test to reflect that https://github.com/shockone/black-screen/blob/master/test/lexer_spec.ts#L22-L22
* (This one is a bit harder) Since we need to pass the exact same input to a shell - otherwise it might have a different meaning - and we changed the lexed input, we can't rely on it anymore. So, we'd want to modify the constructor https://github.com/shockone/black-screen/blob/master/src/PTY.ts#L14-L14 to take a single string parameter instead of the first two, and pass it `job.prompt.value`. There are a few instantiations, which would need to be changed to.

Please let me know if it makes sense and if you're willing to try doing it.

**Comment by adchilds** [comment]
Sure, I'll take a look at that within the next day. Thanks for the input!

**Comment by shockone** [comment]
I just realized that the third step wouldn't work: then it wouldn't expand aliases, history substitution and things like that. Sorry about that, I hope you haven't started your work yet. I think I'll fix it by myself.

**Pull request closed without merging**